### PR TITLE
ttyrec: fix build on Darwin

### PR DESCRIPTION
**Pull request by therealpxc**
###### Motivation for this change
ttyrec doesn't build on Darwin

###### Things done

- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] Linux
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/tools/misc/ttyrec/default.nix [hunk 1]
```diff
@@ -11,7 +11,7 @@ stdenv.mkDerivation rec {
 
   patches = [ ./clang-fixes.patch ];
 
-  makeFlags = [ "CFLAGS=-DSVR4" ]
+  makeFlags = stdenv.lib.optional buildPlatform.isLinux "CFLAGS=-DSVR4"
```
Comment (copumpkin) [inline pkgs/tools/misc/ttyrec/default.nix]: Is this really a `buildPlatform` thing? cc @Ericson2314 who knows best 😄

Comment (copumpkin) [inline pkgs/tools/misc/ttyrec/default.nix]: I _think_ it should be `hostPlatform` though.

Comment (Ericson2314) [inline pkgs/tools/misc/ttyrec/default.nix]: When in doubt, `hostPlatform`.

Comment (therealpxc) [inline pkgs/tools/misc/ttyrec/default.nix]: Sorry I mixed this up, guys. I'll reread the docs before the next time I try to use it.

Comment (Ericson2314) [inline pkgs/tools/misc/ttyrec/default.nix]: @therealpxc I am mainly pleased people are figuring out about `*Platform` at all :). This is new to most, so no worries.

[Also, one can now do `stdenv.*Platform.is*`, if you don't want to add an import.

**Pull request merged**